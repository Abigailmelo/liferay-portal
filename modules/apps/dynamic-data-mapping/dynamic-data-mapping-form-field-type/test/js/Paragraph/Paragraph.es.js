/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import Paragraph from 'source/Paragraph/Paragraph.es';

let component;
const spritemap = 'icons.svg';

const defaultParagraphConfig = {
	name: 'textField',
	spritemap
};

describe('Field Paragraph', () => {
	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should be readOnly', () => {
		component = new Paragraph({
			...defaultParagraphConfig,
			readOnly: true
		});

		expect(component).toMatchSnapshot();
	});

	it('should have an id', () => {
		component = new Paragraph({
			...defaultParagraphConfig,
			id: 'ID'
		});

		expect(component).toMatchSnapshot();
	});

	it('should have a label', () => {
		component = new Paragraph({
			...defaultParagraphConfig,
			label: 'label'
		});

		expect(component).toMatchSnapshot();
	});

	it('should have a placeholder', () => {
		component = new Paragraph({
			...defaultParagraphConfig,
			placeholder: 'Placeholder'
		});

		expect(component).toMatchSnapshot();
	});

	it('should not be required', () => {
		component = new Paragraph({
			...defaultParagraphConfig,
			required: false
		});

		expect(component).toMatchSnapshot();
	});

	it('should render Label if showLabel is true', () => {
		component = new Paragraph({
			...defaultParagraphConfig,
			label: 'text',
			showLabel: true
		});

		expect(component).toMatchSnapshot();
	});

	it('should have a spritemap', () => {
		component = new Paragraph(defaultParagraphConfig);

		expect(component).toMatchSnapshot();
	});

	it('should have a value', () => {
		component = new Paragraph({
			...defaultParagraphConfig,
			value: 'value'
		});

		expect(component).toMatchSnapshot();
	});

	it('should have a key', () => {
		component = new Paragraph({
			...defaultParagraphConfig,
			key: 'key'
		});

		expect(component).toMatchSnapshot();
	});
});
