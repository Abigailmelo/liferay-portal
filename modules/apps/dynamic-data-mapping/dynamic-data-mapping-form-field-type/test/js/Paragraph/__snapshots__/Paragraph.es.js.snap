// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Paragraph should be readOnly 1`] = `
<div class="form-group" data-field-name="textField">
  <div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="textField">
    <div class="liferay-ddm-form-field-paragraph-text"></div>
  </div>
</div>
`;

exports[`Field Paragraph should have a key 1`] = `
<div class="form-group" data-field-name="textField">
  <div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="textField">
    <div class="liferay-ddm-form-field-paragraph-text"></div>
  </div>
</div>
`;

exports[`Field Paragraph should have a label 1`] = `
<div class="form-group" data-field-name="textField"><label>label </label>
  <div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="textField">
    <div class="liferay-ddm-form-field-paragraph-text"></div>
  </div>
</div>
`;

exports[`Field Paragraph should have a placeholder 1`] = `
<div class="form-group" data-field-name="textField">
  <div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="textField">
    <div class="liferay-ddm-form-field-paragraph-text"></div>
  </div>
</div>
`;

exports[`Field Paragraph should have a spritemap 1`] = `
<div class="form-group" data-field-name="textField">
  <div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="textField">
    <div class="liferay-ddm-form-field-paragraph-text"></div>
  </div>
</div>
`;

exports[`Field Paragraph should have a value 1`] = `
<div class="form-group" data-field-name="textField">
  <div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="textField">
    <div class="liferay-ddm-form-field-paragraph-text"></div>
  </div>
</div>
`;

exports[`Field Paragraph should have an id 1`] = `
<div class="form-group" data-field-name="textField">
  <div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="textField">
    <div class="liferay-ddm-form-field-paragraph-text"></div>
  </div>
</div>
`;

exports[`Field Paragraph should not be required 1`] = `
<div class="form-group" data-field-name="textField">
  <div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="textField">
    <div class="liferay-ddm-form-field-paragraph-text"></div>
  </div>
</div>
`;

exports[`Field Paragraph should render Label if showLabel is true 1`] = `
<div class="form-group" data-field-name="textField"><label>text </label>
  <div class="form-group liferay-ddm-form-field-paragraph" data-fieldname="textField">
    <div class="liferay-ddm-form-field-paragraph-text"></div>
  </div>
</div>
`;
